api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/oauth/CsrfCookieProperties.java

```diff
@@ -0,0 +1,21 @@
+package org.apereo.cas.configuration.model.support.oauth;
+
+import org.apereo.cas.configuration.model.support.cookie.CookieProperties;
+
+/**
+ * Setting default values to the pac4j defaults.
```

Comment:
Please add author tags, and also I would rewrite the comments.

For example:

>The pac4j library does not allow for customizing the name or the pinToSession.

This is inaccurate. The pac4j library has nothing to do with pinToSession. Perhaps you want to refer to the CAS component disallowing customization of that setting.

Also, rather than explaining this in the comments, we should reorganize the properties so only relevant properties are inherited by the right classes here. If `CsrfCookieProperties` should not be allowed to use `pinToSession`, then it should not receive that setting from the parent class. Otherwise, the setting shows up in metadata and in the docs and would just be confusing and one extra thing to maintain or remember.

Also please never refer to settings in comments as a String. Use code or link tags. If the setting is ever removed or moved, it's impossible to catch refs like this.

Reply:
In introduced PinnableCookieProperties to hold the pinToSession property. All the existing cookies extend that but there is one `GoogleAnalyticsCookieProperties` that calls `setPinToSession(false)` in its constructor. Not sure if that means it can't ever be true and if it shouldn't ever be true, it should extend `CookieProperites` directly.

Reply:
Looks good. `GoogleAnalyticsCookieProperties` should never have anything to do with session pinning. It should likely extend `CookieProperites`.

Reply:
Done